provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/CachingKotlinCompiler.kt

```diff
@@ -101,16 +100,17 @@ class CachingKotlinCompiler(
 
     data class CompiledPluginsBlock(val lineNumber: Int, val compiledScript: CompiledScript)
 
-    fun compileBuildScript(
+    fun compileGradleScript(
         scriptPath: String,
         script: String,
+        scriptTemplate: KClass<out Any>,
```

Comment:
⭕️  I think the `scriptTemplate` deserves to be the first in the argument list like it is elsewhere.

Reply:
Done